aws_lambda_powertools/utilities/idempotency/persistence/dynamodb.py

```diff
@@ -20,6 +20,8 @@ def __init__(
         self,
         table_name: str,
         key_attr: str = "id",
+        key_attr_value: str = "powertools#idempotency",
```

Comment:
@heitorlessa, I created this as a Draft PR to get your early feedback on the interface. Once we agree on the ergonomics, I'm happy to update documentation and add examples.

Reply:
@heitorlessa, I didn't see any feedback so I moved this from Draft to Ready. I would still like to update documentation and examples before merging this PR, and would love some feedback on the API before I take the time to do that.

Reply:
Because DynamoDB IAM resource conditions don't allow you to restrict row level access based on SK, where the function name is known, I'd suggest we change the primary key slightly to: `idempotency#function-name`.

If they want, for example, to restrict anyone from ever modifying idempotency tokens for X function that has a **prod** stage or something more specific to their business in the name they now can.

Here are the scenarios I understood by going through the PR, as **C** is new - please correct me if I misunderstood them.

---

## Scenarios

**A) No hash key set.:** Use default hash key and value

```python
from aws_lambda_powertools.utilities.idempotency import DynamoDBPersistenceLayer

# KEY_ATTR="id"
# KEY_ATTR_VALUE="test-func#ec534ff6b4746107270b412333b59f52"
persistence_layer = DynamoDBPersistenceLayer(table_name="IdempotencyTable")

```

**B) Hash key set.:** Use new hash key and default value

```python
# KEY_ATTR="idempotency_key"
# KEY_ATTR_VALUE="test-func#ec534ff6b4746107270b412333b59f52"
persistence_layer = DynamoDBPersistenceLayer(table_name="IdempotencyTable", key_attr="idempotency_key")
```

**C) Hash and Sort key set.:** Use new hash and sort key, and default value as sort key 

```python
# KEY_ATTR="pk"
# SORT_KEY_ATTR="sk"
# KEY_ATTR_VALUE="idempotency#test-func"
# SORT_KEY_ATTR_VALUE="test-func#ec534ff6b4746107270b412333b59f52"
persistence_layer = DynamoDBPersistenceLayer(
    table_name="IdempotencyTable", 
    key_attr="idempotency_key",
    sort_key_attr="idempotency_key")
```

Reply:
@heitorlessa, re the scenarios: yes, C is new. I think there is some copy/pasta in the code, the `key_attr` and `sort_key_attr` should match the comments, as in:

```python
# KEY_ATTR="pk"
# SORT_KEY_ATTR="sk"
# KEY_ATTR_VALUE="idempotency#test-func"
# SORT_KEY_ATTR_VALUE="test-func#ec534ff6b4746107270b412333b59f52"
persistence_layer = DynamoDBPersistenceLayer(
    table_name="IdempotencyTable", 
    key_attr="pk",       # <- changed to match comment
    sort_key_attr="sk")  # <- changed to match comment
```

There is also a related scenario in which the `DynamoDBPersistenceLayer` passes the desired value for `key_attr_value`, e.g.:

```python
# KEY_ATTR="pk"
# SORT_KEY_ATTR="sk"
# KEY_ATTR_VALUE="known-static-partition-key-value"
# SORT_KEY_ATTR_VALUE="test-func#ec534ff6b4746107270b412333b59f52"
persistence_layer = DynamoDBPersistenceLayer(
    table_name="IdempotencyTable", 
    key_attr="pk",
    key_attr_value="known-static-partition-key-value",       
    sort_key_attr="sk")
```